tests/Feature/AnalyticsTest.php

```diff
@@ -129,4 +129,59 @@ public function a_page_view_from_an_excluded_ip_is_not_tracked_if_enabled()
             'uri' => '/test',
         ]);
     }
+
+    /** @test */
+    public function utm_details_can_be_saved_with_page_views()
+    {
+        $request = Request::create('/test', 'GET', [
+            'utm_source' => 'test-source',
+            'utm_medium' => 'test-medium',
+            'utm_campaign' => 'test-campaign',
+            'utm_term' => 'test-term',
+            'utm_content' => 'test-content',
+        ]);
+        $request->setLaravelSession($this->app['session']->driver());
+
+        (new Analytics())->handle($request, function ($req) {
+            $this->assertEquals('test', $req->path());
+            $this->assertEquals('GET', $req->method());
+        });
+
+        $this->assertCount(1, PageView::all());
+        $this->assertDatabaseHas('page_views', [
+            'uri' => '/test',
+            'device' => 'desktop',
+            'utm_source' => 'test-source',
+            'utm_medium' => 'test-medium',
+            'utm_campaign' => 'test-campaign',
+            'utm_term' => 'test-term',
+            'utm_content' => 'test-content',
+        ]);
+    }
+
+    /** @test */
+    public function utm_details_will_be_trimmed()
+    {
+        $string = 'Nv19I4yx6b8OEjFhtSqYGANwRim0WJHTQoUDaK3vc72Xl5rMZk1PpFLBdVuCfZgs7TwMeRhxYi9n6CLt2pzqOBGjXSaHwvyJ8KEVNv19I4yx6b8OEjFhtSqYGANwRim0WJHTQoUDaK3vc72Xl5rMZk1PpFLBdVuCfZgs7TwMeRhxYi9n6CLt2pzqOBGjXSaHwvyJ8KEVNv19I4yx6b8OEjFhtSqYGANwRim0WJHTQoUDaK3vc72Xl5rMZk1PpFLBdVuCfZgs7TwMeRhxYi9n6CLt2pzqOBGjXSaHwvyJ8KEV';
```

Comment:
Any chance we could use a helper function or something to generate this string or does it need to be this specific one for any specific reason?

Reply:
No reason in particular. I've updated it to use the string helper